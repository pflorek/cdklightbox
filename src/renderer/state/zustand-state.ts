import create from 'zustand';
import { persist, subscribeWithSelector } from 'zustand/middleware';
import { Set, WorkbenchState } from './states';
import { errorState } from './substates/error-state';
import { workingDirectoryState } from './substates/working-directory-state';

import { statusState } from './substates/status-state';
import { cdkAppState } from './substates/cdk-app-state';

// eslint-disable-next-line import/prefer-default-export
export const useWorkbenchStore = create<WorkbenchState>(
  // @ts-ignore
  subscribeWithSelector(
    persist(
      (set: Set) => ({
        ...errorState(set),
        ...statusState(set),
        ...workingDirectoryState(set),
        ...cdkAppState(set),
      }),

      {
        name: 'localStorage',
        getStorage: () => localStorage,
      }
    )
  )
);
