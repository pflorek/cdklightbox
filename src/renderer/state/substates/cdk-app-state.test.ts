import * as path from 'path';
import * as fs from 'fs';
import { useWorkbenchStore } from '../zustand-state';

describe('cdk-app-state', () => {
  it('children are flattened', () => {
    const tree = JSON.parse(
      fs.readFileSync(path.join(__dirname, 'test_states', 'tree.json'), 'utf-8')
    );
    useWorkbenchStore.getState().setCdkApp(tree);

    expect(
      useWorkbenchStore.getState().cdkApp!.flattenedChildren
    ).toMatchSnapshot();
  });
});
