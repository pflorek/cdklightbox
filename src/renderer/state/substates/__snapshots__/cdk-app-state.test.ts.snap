// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cdk-app-state children are flattened 1`] = `Array []`;
